packages/framework/fluid-static/src/fluidContainer.ts

```diff
@@ -202,9 +223,10 @@ export class FluidContainer extends TypedEventEmitter<IFluidContainerEvents> imp
     }
 
     /**
-     * {@inheritDoc IFluidContainer.attach}
+     * {@inheritDoc IFluidContainer.connect}
      */
     public async attach(): Promise<string> {
+        // TODO: this needs to be implemented. This type should be self-sufficient.
```

Comment:
This is a good note to add but we should probably take a bit deeper look at the implementation here to see if we can do better. This is using some bad practices to over come the problem that the creator of this class is the only one with the knowledge to implement the response here.

This is because we are trying to force separation of the container service/state from the FluidContainer object itself. The problem is that the underlying implementation doesn't have this guarantee at the moment.

```suggestion
        // This needs to be implemented/overwritten by the creator of this object. The reason is because externally we are presenting a separation between the service and the FluidContainer but internally this separation is not there.
```

Reply:
@ChumpChief since this relates to your earlier comment.

It seems like we need to find consensus on how this type is intended to be consumed. It sounds like the long-term desire is for this to be a self-sufficient type, but for now it's a class masquerading as an abstract class (and currently without the type-docs making this clear to consumers).

If we think there is a plan to clean up the service/container separation in the near-term, then adding a few more comments about current requirements is probably sufficient. If we don't, I think it would be worth making this an abstract class (at least in the short term) to make expected requirements of consumers more clear.

Reply:
There are two consumers of this class. The C1 end developer who's calling attach and ourselves (in the form of the *-client packages) who needs to implement it on a per service basis.

I'm okay with any solution as long as we prioritize clear API documentation for the C1. If there's an good way to do that with api extractor then sweet. Otherwise I would prefer the function docs to be clear to the C1 and add an inline comment for the developer.